test/client.jl

```diff
@@ -13,6 +13,7 @@ using InteractiveUtils: @which
 
 # test we can adjust default_connection_limit
 HTTP.set_default_connection_limit!(12)
+@test HTTP.Connections.TCP_POOL[].max == 12
```

Comment:
i think ideally we'd also reset the limit a second time and check it take effect (who knows, maybe `12` is/will be the default one day), and test all the pools

how about:
```suggestion
for x in (10, 12)
    HTTP.set_default_connection_limit!(x)
    @test HTTP.Connections.TCP_POOL[].max == x
    @test HTTP.Connections.MBEDTLS_POOL[].max == x
    @test HTTP.Connections.OPENSSL_POOL[].max == x
end
```
?

Reply:
Great idea